### ISSUE TITLE
Mismatch of implement and comment of attention_mask  in clip encoder?

### ISSUE THREAD

**Issue by hust-nj** [huggingface/transformers#31726]
I found the comment in CLIPEncoder

https://github.com/huggingface/transformers/blob/e65502951593a76844e872fee9c56b805598538a/src/transformers/models/clip/modeling_clip.py#L592-L596

mismatch with the code implement here

https://github.com/huggingface/transformers/blob/e65502951593a76844e872fee9c56b805598538a/src/transformers/models/clip/modeling_clip.py#L296

**Comment by NielsRogge** [comment]
Hi,

The `attention_mask` gets internally turned into a tensor containing either zeros or very low values https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_attn_mask_utils.py

**Comment by amyeroberts** [comment]
Hi @hust-nj, thanks for opening an issue! 

Is the "mismatch" you're highilghting regarding the values or the shapes? As @NielsRogge mentions, for the attention calculation, the values are converted to floats and values will become large negative numbers or zeros. The masks are also expanded [within the forward pass](https://github.com/huggingface/transformers/blob/e65502951593a76844e872fee9c56b805598538a/src/transformers/models/clip/modeling_clip.py#L702) to create a 4d mask

**Comment by hust-nj** [comment]
> Hi @hust-nj, thanks for opening an issue!
> 
> Is the "mismatch" you're highilghting regarding the values or the shapes? As @NielsRogge mentions, for the attention calculation, the values are converted to floats and values will become large negative numbers or zeros. The masks are also expanded [within the forward pass](https://github.com/huggingface/transformers/blob/e65502951593a76844e872fee9c56b805598538a/src/transformers/models/clip/modeling_clip.py#L702) to create a 4d mask

I mean the value of mask in clip `vision` model(not the text part) ranging from 0 to 1 or  0 to -float(inf)

**Comment by amyeroberts** [comment]
@hust-nj The clip vision model [doesn't take an attention mask](https://github.com/huggingface/transformers/blob/e65502951593a76844e872fee9c56b805598538a/src/transformers/models/clip/modeling_clip.py#L829).

**Comment by hust-nj** [comment]
Got it, thank you 

> @hust-nj The clip vision model [doesn't take an attention mask](https://github.com/huggingface/transformers/blob/e65502951593a76844e872fee9c56b805598538a/src/transformers/models/clip/modeling_clip.py#L829).
